internal/worker_runner.go

```diff
@@ -144,24 +153,40 @@ func (wr *workerRunner) composeDeviceEvent(powerStatus api.PowerStatusResponse,
 	}
 	// only update location if no error
 	if locationErr == nil {
-		statusData.Location = location
+		statusData.Vehicle.Signals = appendSignalData(statusData.Vehicle.Signals, "longitude", location.Longitude)
+		statusData.Vehicle.Signals = appendSignalData(statusData.Vehicle.Signals, "latitude", location.Latitude)
+		statusData.Vehicle.Signals = appendSignalData(statusData.Vehicle.Signals, "hdop", location.Hdop)
+		statusData.Vehicle.Signals = appendSignalData(statusData.Vehicle.Signals, "nsat", location.Nsat)
 	}
-	n := &models.Network{}
 
 	// only update Wi-Fi if no error
 	if wifiErr == nil {
-		n.WiFi = *wifi
-		statusData.Network = n
-	}
-	// only update cell info if no error
-	if cellErr == nil {
-		n.QMICellInfoResponse = cellInfo
-		statusData.Network = n
+		statusData.Vehicle.Signals = appendSignalData(statusData.Vehicle.Signals, "wpa_state", wifi.WPAState)
```

Comment:
does wifi stuff fall under "network" category or did you guys decide it should be separate from cell network

Reply:
I believe we dcided that wifi should be separate from the network. @KevinJoiner please chime in

Reply:
We wanted to include Wi-Fi information with the vehicle status. However, I think we may want to duplicate the Wi-Fi information on both the vehicle status and network status since it is still relevant network information.

Reply:
Make sense to me, see update on it

Reply:
```
{
  "id":"2es9GNZYbKRpS34IyZhZp8ZBfAO",
  "source":"aftermarket/device/network",
  "specversion":"1.0",
  "subject":"0x98D78d711C0ec544F6fb5d54fcf6559CF41546a9",
  "time":"2024-04-09T17:05:21.620769984Z",
  "type":"com.dimo.device.network",
  "signature":"0x70b02f727277b20433672da4f0108c0f1193c594784070b980a9abd1bac7605d3ca6b39308d5e8778c4e2dbb07a344e3219d10c994e177edfc4729b597589d671c",
  "data":{
    "timestamp":1712682321620,
    "cell":{
      "lte_info_neighboring_gsm":{
        "ue_in_idle":"yes"
      },
      "interfrequency_lte_info":{
        "ue_in_idle":"yes"
      },
      "intrafrequency_lte_info":{
        "cell_[1]":{
          "rssi":"",
          "physical_cell_id":0,
          "rsrp":"",
          "rsrq":""
        },
        "plmn":310026,
        "global_cell_id":12456455,
        "ue_in_idle":"yes",
        "tracking_area_code":22002,
        "cell_[0]":{
          "rssi":"-75.0 dBm",
          "physical_cell_id":53,
          "rsrp":"-104.9 dBm",
          "rsrq":"-8.9 dB"
        },
        "eutra_absolute_rf_channel_number":"875 (E-UTRA band 2: 1900 PCS)",
        "serving_cell_id":53,
        "cell_[2]":{
          "rssi":"",
          "physical_cell_id":0,
          "rsrp":"",
          "rsrq":""
        }
      },
      "lte_info_neighboring_wcdma":{
        "ue_in_idle":"yes"
      }
    },
    "wifi":{
      "wpa_state":"COMPLETED",
      "ssid":"Fios-Vn2gM_EXT"
    }
  }
}
```